### ISSUE TITLE
Does it support streaming

### ISSUE THREAD

**Issue by dengcunqin** [youngsheen/SimVQ#7]
Does it support streaming

**Comment by youngsheen** [comment]
We used the same architecture as wavtokenizer. There are two components that limit streaming: the CNN and the attention module. By default, wavtokenizer does not enable causal mode. In our experiments, we applied causal mode to the CNN but overlooked the attention part. Fortunately, the encoder does not contain attention but decoder has.

If you want a truly streaming model, you may need to retrain a version with causal attention. We also consider retraining a streaming version in the future.

**Comment by VJJJJJJ1** [comment]
> We used the same architecture as wavtokenizer. There are two components that limit streaming: the CNN and the attention module. By default, wavtokenizer does not enable causal mode. In our experiments, we applied causal mode to the CNN but overlooked the attention part. Fortunately, the encoder does not contain attention but decoder has.
> 
> If you want a truly streaming model, you may need to retrain a version with causal attention. We also consider retraining a streaming version in the future.

Did you just modify the causal=True for the encoder?

**Comment by youngsheen** [comment]
I modify the CNN modules in the encoder and decoder to causal mode.

**Comment by VJJJJJJ1** [comment]
> I modify the CNN modules in the encoder and decoder to causal mode.

I mean, did you just set causal to True in calss SConv1d, or did you make additional modifications to the CNN? Thank you for your reply!

**Comment by youngsheen** [comment]
Just set causal to true without any other additional modifications.

**Comment by VJJJJJJ1** [comment]
> Just set causal to true without any other additional modifications.

thank u

**Comment by VJJJJJJ1** [comment]
> Just set causal to true without any other additional modifications.

BTW, how do you modify `self.embed = nn.Conv1d(input_channels, dim, kernel_size=7, padding=3)` in `SimVQ/tree/main/taming/modules/diffusionmodules
/vocosbackbone.py`, I tried as follows: ` self.embed = SConv1d(input_channels, dim, kernel_size=7, causal=True)`
is it correct?
